src/main/java/mesosphere/marathon/core/leadership/CandidateImpl.java

```diff
@@ -149,8 +175,9 @@ public CandidateImpl(
                                 membership.getMemberPath(), memberIds));
 
                         leader.onElected(new ExceptionalCommand<Group.JoinException>() {
-                            @Override public void execute() throws Group.JoinException {
-                                membership.cancel();
+                            @Override
+                            public void execute() throws Group.JoinException {
+                                membershipRef.get().cancel(); // here we know that membershipRef != null
```

Comment:
Can't [L199](https://github.com/mesosphere/marathon/pull/3732/commits/5c1d11e1d13e4646b2b4bfa48a8afa9e080ed19e?diff=split#diff-2b73a7298e29c2c5b15b8f7aae5827bfR199) be executed before this line?

For example if leader loses leadership and the abdication command is called at roughly the same time?